test/e2e/gameserver_test.go

```diff
@@ -522,17 +525,31 @@ func TestGameServerWithPortsMappedToMultipleContainers(t *testing.T) {
 	defer framework.AgonesClient.AgonesV1().GameServers(defaultNs).Delete(readyGs.ObjectMeta.Name, nil) // nolint: errcheck
 	assert.Equal(t, readyGs.Status.State, agonesv1.GameServerStateReady)
 
-	firstContainerReply, err := e2eframework.SendGameServerUDPToPort(readyGs, "gameport", "Ping 1")
-	if err != nil {
-		t.Fatalf("Could not message GameServer: %v", err)
-	}
-	assert.Contains(t, firstContainerReply, "Ping 1")
+	interval := 2 * time.Second
+	timeOut := 60 * time.Second
 
-	secondContainerReply, err := e2eframework.SendGameServerUDPToPort(readyGs, "second-gameport", "Ping 2")
-	if err != nil {
-		t.Fatalf("Could not message GameServer: %v", err)
+	expectedMsg1 := "Ping 1"
+	errPoll := wait.PollImmediate(interval, timeOut, func() (done bool, err error) {
+		res, err := e2eframework.SendGameServerUDPToPort(readyGs, firstPort, expectedMsg1)
+		if err != nil {
+			t.Logf("Could not message GameServer on %s: %v. Will try again...", firstPort, err)
+		}
+		return err == nil && strings.Contains(res, expectedMsg1), nil
+	})
+	if errPoll != nil {
+		assert.FailNow(t, errPoll.Error(), "expected no errors after polling a port: %s", firstPort)
 	}
-	assert.Contains(t, secondContainerReply, "Ping 2")
+
+	expectedMsg2 := "Ping 2"
+	errPoll = wait.PollImmediate(interval, timeOut, func() (done bool, err error) {
+		res, err := e2eframework.SendGameServerUDPToPort(readyGs, secondPort, expectedMsg2)
+		if err != nil {
+			t.Logf("Could not message GameServer on %s: %v. Will try again...", secondPort, err)
+		}
+		return err == nil && strings.Contains(res, expectedMsg2), nil
+	})
+
+	assert.Nil(t, errPoll, "expected no errors after polling a port: %s", secondPort)
```

Comment:
```suggestion
	assert.NoError(t, errPoll, "expected no errors after polling a port: %s", secondPort)
```

Just displays the details of the passed in error message. Other than that, this looks good to go :+1:

Reply:
Done